marklogic-rdf4j/src/test/java/com/marklogic/semantics/rdf4j/MarkLogicRepositoryConnectionTest.java

```diff
@@ -662,6 +660,9 @@ public void testHasStatementBNode() throws Exception
         Statement st1 = vf.createStatement(alice, name, alicesName);
         conn.add(st1, context1);
         Assert.assertTrue(conn.hasStatement(null, null, alicesName, false));
+
+        BNode markName = vf.createBNode();
+        Assert.assertFalse(conn.hasStatement(null, null, markName, false));
```

Comment:
Since markName is in effect a free variable, this assertion should be true iff there's at least one triple in the database.
In actuality, your implementation returns false because markName will never match a previously-inserted bnode.